src/domain/user.ts

```diff
@@ -0,0 +1,6 @@
+export class User {
```

Comment:
nit: seems overly OOP to have a class for something that could just be a well-typed object like 
```
export interface User {
  username: string;
  email string;
}
```

Reply:
Good call. Fixed.